woocommerce/woocommerce-android
# Issue/11559 product sales last 30 days

Part II of : #11559 
Part I was implemented and merged here: https://github.com/woocommerce/woocommerce-android/pull/11582


### Description

In part I https://github.com/woocommerce/woocommerce-android/pull/11582 we added a new card to display the product stock. In this PR I'm adding the missing logic to display the product sales for the last 30 days. 

![Screenshot 2024-05-30 at 23 29 48](https://github.com/woocommerce/woocommerce-android/assets/2663464/aa7afe0f-73f7-4c88-9166-d5687360c15e)

In order to display product sales we need to make

2 Missing parts in this PR that I'll tackle in subsequent PRs: 
- Disabling `Stock` card if the store doesn't have any products or orders yet. Currently, if the store doesn't have any product or orders in place and the stock card is added to the dashboard the `error` state is displayed. So this is a good scenario for testing the error state of the card. 
- `ProductStockRepository` unit tests 
- If the product is a variation, upon clicking on it, navigate directly to the variation instead of the parent product. I made a quick attempt to implement this by directly passing the `variationId` to the product detail navigation event, but didn't work as expected. A lot of information was missing when the variation was opened. Will look into this further. 

### Testing instructions


1. Log into a site that has products, product variations, and orders placed in the last 30 days. 
2. Edit the stock of the different product and set some to have low stock (1 or 2 units left), some out of stock, and some in back order. 
3. Add `Stock` card to the dashboard
4. Change between the different stock statuses from the card filter and verify the data information is correct.
